modules/collect/src/main/java/com/opengamma/collect/timeseries/DenseLocalDateDoubleTimeSeries.java

```diff
@@ -0,0 +1,984 @@
+        int endPointAdjustments = remaining < (6 - startDate.getDayOfWeek().getValue()) ? remaining : remaining + 2;
+        return startDate.plusDays(7 * numWeekEnds + endPointAdjustments);
+      }
+    },
+    /**
+     * Data is held for weekends.
+     */
+    INCLUDE_WEEKENDS {
+      @Override
+      int calculatePosition(LocalDate startDate, LocalDate toInsert) {
+        return (int) DAYS.between(startDate, toInsert);
+      }
+
+      @Override
+      LocalDate calculateDateFromPosition(LocalDate startDate, int position) {
+        return startDate.plusDays(position);
+      }
+    };
+
+    /**
+     * Calculates the position in the array where the supplied date should
+     * be located given a start date. As no information is held about the
+     * actual array, callers must check array bounds.
+     *
+     * @param startDate  the start date for the series (the value for this
+     *   entry will be stored at position 0 in the array)
+     * @param date  the date to calculate a position for
+     * @return the position in the array where the date would be located
+     */
+    abstract int calculatePosition(LocalDate startDate, LocalDate date);
+
+    /**
+     * Given a start date and a position in an array, calculate what date
+     * the position holds data for.
+     *
+     * @param startDate  the start date for the series (the value for this
+     *   entry will be stored at position 0 in the array)
+     * @param position  the position in the array to calculate a date for
+     * @return the date the position in the array holds data for
+     */
+    abstract LocalDate calculateDateFromPosition(LocalDate startDate, int position);
+  }
+
+  /**
+   * Date corresponding to first element in the array. All other
+   * values can be calculated using date arithmetic to find
+   * correct point
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate startDate;
+
+  @PropertyDefinition(get = "private", validate = "notNull")
+  private final double[] points;
+
+  /**
+   * Whether we should store data for the weekends (NaN will be stored
+   * if no data is available).
+   */
+  @PropertyDefinition(get = "private", validate = "notNull")
+  private final DenseTimeSeriesCalculation dateCalculation;
+
+  /**
+   * Obtains a time-series from a collection of points.
+   * The points must not contain weekend dates. If weekend
+   * dates are required then use
+   * {@link #of(Collection, DenseTimeSeriesCalculation)}.
+   *
+   * @param points  the list of points
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(Collection<LocalDateDoublePoint> points) {
+    return of(points, SKIP_WEEKENDS);
+  }
+
+  /**
+   * Obtains a time-series from a collection of points
+   * using the specified {@code dateCalculation}.
+   *
+   * @param points  the list of points
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(
```

Comment:
`dateCalculation` missing `@param`. Seems like it should be an internal detail, not a public factory parameter. If the data has weekends then just store in that way.